tests/tx/test_tx.py

```diff
@@ -517,7 +517,6 @@ def test_tx_methods(self):
 
         # Validate is_genesis without storage
         tx_equal.storage = None
-        self.assertFalse(tx_equal.is_genesis)
```

Comment:
Why did you remove it? Maybe we should just check if this raises `ValueError`

Reply:
The `is_genesis` is a method marked as `@property`. It does not work anymore without a storage.

Reply:
Yes, that's why I said we could check for a `ValueError` because that's what you raise when there is no storage.

Reply:
Why does it need a storage? We should be able to tell from looking at the settings only.

Reply:
One of the ideas of this PR is to use the storage as the point of truth regarding genesis. Maybe we can create a `is_genesis(tx_id: bytes)` method in `hathor.transaction.genesis`, and use it directly.

Reply:
While I think it's generally useful to contain the concerns, even the storage will have to rely on the settings to know what are the genesis transactions. So I don't think it's useful to "prevent" these settings from being useful outside the storage.

I'd say that the storage can definitely be the gatekeeper of all metadata, but determining if a transaction is part of the genesis isn't inherently something that should depend on the metadata.

 And because there actually are use cases (like this assert or `calculate_height`'s `if`) that are prevented from using `is_genesis` because the lack of a storage would break it. I think it's more useful not to use the storage when it makes sense.

Reply:
I agree. I reverted this change. Would you review it again, please?

Reply:
Good to me

Reply:
Great!